src/api/firebase.js

```diff
@@ -189,12 +189,32 @@ export async function addItem(listPath, { itemName, daysUntilNextPurchase }) {
 	});
 }
 
-export async function updateItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to update an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+export async function updateItem(listPath, item) {
+	// creates a reference to the item collection
+	const listCollectionRef = collection(db, listPath, 'items');
+	// creates a reference to current item by passing the item's id
+	const itemRef = doc(listCollectionRef, item.id);
+
+	// updating the item in the item's collection by passing it the itemRef
+	await updateDoc(itemRef, {
+		// creates a new key value pair named "isChecked" and sets it with the isChecked state value
+		isChecked: item.isChecked,
```

Comment:
Hi! Can you help me understand why we're storing `isChecked`? It looks like that state is determined by either the user toggling the checkbox or based the `dateLastPurchased`.

Reply:
Hey Belinda, 
We created the new key in the items object called `isChecked`  and thought it would be a good approach to store the value of the `isChecked` state here, that way we would update `dateLastPurchased` and `totalPurchases` based on the checkbox state. Having the key there allowed us to visually confirm in Firestore if the checkbox was `true` or `false.

We felt like this synchronized the UI and Firestore database. But if there is a better approach we are open to discuss/refeactor!